src/modules/generateRandomNumber.js

```diff
@@ -7,7 +7,26 @@
  * @return {number} A random 4-digit number
  */
 function generateRandomNumber() {
-  /* Write your code here */
+  let randomNumber = '';
+  const previousNumbers = [];
+
+  for (let i = 0; i < 4;) {
+    const randomNum = Math.floor(Math.random() * 10);
+
+    if (randomNum === 0 && i === 0) {
+      continue;
+    }
+
+    if (previousNumbers.includes(randomNum)) {
+      continue;
+    }
+
+    previousNumbers.push(randomNum);
+    randomNumber += randomNum.toString();
+    i++;
+  }
+
+  return +randomNumber;
```

Comment:
This block can be modified in the next way
```suggestion
function generateRandomNumber() {
  const secretCode = Array.from({ length: 4 }, (el) => 'x');

  for (let i = 0; i < secretCode.length; i++) {
    secretCode[i] = assignRandomNumber(secretCode);
  }

  return Number(secretCode.join(''));
}

const assignRandomNumber = (secretCode) => {
  const number = Math.floor(Math.random() * 9 + 1);

  if (secretCode.includes(number)) {
    return assignRandomNumber(secretCode);
  }

  return number;
};
```